apps/burn/burn.js

```diff
@@ -0,0 +1,254 @@
+  }
+}
+
+// Function to write the counter to the file
+function writeCounterToFile() {
+  // Open the file in write mode
+  file = Storage.open(PATH, "w");
+  // Write the counter and date to the file
+  file.write(counter.count.toString() + "," + counter.date + "\n");
+}
+
+// Function to reset the counter
+function resetCounter() {
+  // Reset the counter to 0 and update the date
+  counter = { count: 0, date: dayString() };
+}
+
+// Function to update the counter value
+function updateCounterValue(value) {
+  // Update the counter with the new value, ensuring it's not less than 0
+  counter = { count: Math.max(0, value), date: dayString() };
+}
+
+// Function to update the counter
+function updateCounter(value) {
+  // If the date has changed, reset the counter
+  if (counter.date != dayString()) {
+    resetCounter();
+  } else {
+    // Otherwise, update the counter value
+    updateCounterValue(value);
+  }
+
+  // Write the updated counter to the file
+  writeCounterToFile();
+  // Update the screen with the new counter value
+  updateScreen();
+}
+
+// Function to set a watch on a button to update the counter when pressed
+function counterButtonWatch(button, increment) {
+  setWatch(
+    () => {
+      // If the button is for incrementing, or the counter is greater than 0, update the counter
+      if (increment || counter.count > 0) {
+        updateCounter(
+          counter.count + (increment ? getInterval() : -getInterval())
+        );
+        // Update the screen with the new counter value
+        updateScreen();
+      }
+    },
+    button,
+    { repeat: true }
+  );
+}
+
+// Function to create interval functions
+const createIntervalFunctions = function () {
+  // Array of intervals
+  const intervals = [50, 100, 200, 10];
+  // Current location in the intervals array
+  let location = 0;
+
+  // Function to get the current interval
+  const getInterval = function () {
+    return intervals[location];
+  };
+
+  // Function to rotate the increment
+  const rotateIncrement = function () {
+    // Update the location to the next index in the intervals array, wrapping around if necessary
+    location = (location + 1) % intervals.length;
+    // Update the screen with the new increment
+    updateScreen();
+  };
+
+  // Return the getInterval and rotateIncrement functions
+  return { getInterval, rotateIncrement };
+};
+
+// Create the interval functions
+const intervalFunctions = createIntervalFunctions();
+const getInterval = intervalFunctions.getInterval;
+const rotateIncrement = intervalFunctions.rotateIncrement;
+
+// Check if the hardware version is Bangle.js 2
+const BANGLEJS2 = process.env.HWVERSION == 2;
+
+// Variables for drag handling and screen coordinates
+let drag, y, x;
+
+// If the hardware version is Bangle.js 2, set the coordinates accordingly
+if (BANGLEJS2) {
+  drag = undefined;
+  y = 45;
+  x = 5;
+} else {
+  // If the hardware version is Bangle.js 1, set the coordinates accordingly
+  y = 100;
```

Comment:
Lint warning: `'y' is assigned a value but never used`

So seems like you could just remove it?